File: se.softhouse.garden.orchid.demo.vaadin/src/main/java/se/softhouse/garden/orchid/demo/vaadin/MSG.java
package se.softhouse.garden.orchid.demo.vaadin;

import static se.softhouse.garden.orchid.spring.text.OrchidMessageSource.code;
import se.softhouse.garden.orchid.spring.text.OrchidMessageConstants;

public class MSG extends OrchidMessageConstants {

	private static MSG in = new MSG();

	public static String HELLO_MESSAGE(String name) {
		return in.msg.get(code("hello.message").arg("name", name));
	}

	public static String COLOR_MESSAGE(String id) {
		return in.msg.get(code("color.message").arg("colorId", id));
	}

	public static String WINDOW_TITLE = in.msg.get("window.title");
}

File: se.softhouse.garden.orchid.scala.demo.vaadin/src/main/scala/se/softhouse/garden/orchid/scala/demo/vaadin/MSG.scala
package se.softhouse.garden.orchid.scala.demo.vaadin;

import javax.annotation.Resource
import se.softhouse.garden.orchid.spring.DI
import se.softhouse.garden.orchid.spring.text.OrchidLocalizedMesageSource;
import se.softhouse.garden.orchid.scala.spring.OrchidMessage

object MSG {

  case class HELLO_MESSAGE(name: String) extends OrchidMessage("hello.message")
  case class COLOR_MESSAGE(colorId: String) extends OrchidMessage("color.message")
  val WINDOW_TITLE = OrchidMessage("window.title")
}
